### PR TITLE
WMS options

### PR DESCRIPTION
**Pull request by ocefpaf**
Work in progress to replace #277 

See: http://nbviewer.jupyter.org/gist/ocefpaf/190dd0792ea05c0e5d3d

### CODE REVIEW

**Comment by BibMartin** [comment]
Happy to see the PR back.

In my mind, we were about to release `v0.2` ; do you want to incorporate this PR in it ?

**Comment by ocefpaf** [comment]
Yep.  I will finish this today and focus on the release.  I don't want to delay it more, but I meed WMS styles there for my own selfish reasons :grin:

**Comment by ocefpaf** [comment]
@BibMartin I removed the legacy WMS method test.  That test was already a little flaky and we will remove method in after `v0.2.0`.  However, I did not add a proper unittest for the new options. 

The current test checks if the HTML is rendered and for the existence of bounds. The bounds are `None` now but I have a plan to use OWSLib to feed that information there.  OWSLib will be an option dependency and I will leave that implementation to `v0.3.0`.

Can you review and merge this one?

File: folium/features.py [hunk 1]
```diff
@@ -23,45 +23,57 @@ class WmsTileLayer(Layer):
...
+        # Options.
+        self.layers = layers if layers else ''
+        self.styles = styles if styles else ''
+        self.format = format if format else 'image/jpeg'
         self.transparent = transparent
-        self.attr = attr
-
+        self.version = version
+        # FIXME: Should be map CRS!
+        # self.crs = crs if crs else 'null
```
Comment (BibMartin) [inline folium/features.py]: Does it mean that you would like to use `this._parent.crs` in the template to set the CRS ?

**Comment by BibMartin** [comment]
OK, go!

File: folium/features.py (hunk 1 above)
Comment (ocefpaf) [inline folium/features.py]: I understood leaflet correct the WMS service might be in a different `crs` than the map and this variable is used to re-project it to the base map.

**Comment by ocefpaf** [comment]
Thanks @BibMartin!  Preparing the  release now...

**Comment by BibMartin** [comment]
Don't you want to finish #352 before ?

**Comment by ocefpaf** [comment]
Working on that too.

**Pull request merged**